src/test/ui/rfc-2565-param-attrs/param-attrs-builtin-attrs.rs

```diff
@@ -0,0 +1,257 @@
+#![feature(param_attrs)]
+
+pub fn foo(
+    #[inline] a: u32,
+    //~^ ERROR allow, cfg, cfg_attr, deny, forbid and warn are the only allowed built-in
+    #[must_use] #[no_mangle] b: i32
+    //~^ ERROR allow, cfg, cfg_attr, deny, forbid and warn are the only allowed built-in
+) {}
+
+// self, &self and &mut self
+
+struct SelfStruct {}
+impl SelfStruct {
+    fn bar(
+        #[inline] self,
```

Comment:
I don't think it's necessary to test forbidden attribute names in all positions, one position would be enough.
The restriction is entirely orthogonal to attribute positions and walls of text are hard to make sense of.

Reply:
The orthogonality is true but this is also assuming how the thing is implemented, which I find to be a rather bad testing methodology. These sort of redundancies in testing is important when the code base changes and you want to prevent regressions. You may not need to test them on every single argument (perhaps the first, middle, and last is good enough in various cases) but every place where an `fn` can be included should be tested in case the parsing stops being uniform.

Reply:
Yes, the *parsing* test (`param-attrs-pass.rs`) should include all these cases, but this is not a parsing test, this is a test checking attribute names.

Reply:
In my view, the logic about redundancy in testing and trying to cover more of the search space is the same regardless of parsing or attribute name checking. In both cases the underlying implementation could change.

Reply:
Since you cannot test `N x N x N x ...` across all dimensions you have always apply intuition and prune tests to the reasonable subset.
In this case my intuition says that the test as is is unreasonable and need to be reduced, which will be especially evident when the attribute name check is moved to AST validation.

Reply:
Sure, I'm not against reducing it, and you don't need to cover `N x N x N` but having at least some subset of N would be good. If you can reduce it in one dimension (e.g. number of formal parameters, and the number of those that have attributes) you should be able to substantially prune while also covering many locations.

Reply:
Ok, I forgot how irregular the fn header visiting infrastructure is.
Perhaps testing everything was a good idea in this case.